src/agent/adu_core_interface/src/device_properties.c

```diff
@@ -9,10 +9,7 @@
 #include "device_properties.h"
 #include <aduc/logging.h>
 #include <aduc/types/update_content.h>
-
-#ifndef ADUC_PLATFORM_SIMULATOR // DO is not used in sim mode
```

Comment:
shouldn't this be a conditional on whether or not we're using DO?  A check is still relevant, but this was the wrong check.

Reply:
That's for another PR. The agent shouldn't care about DO at all. Only DO downloader does. But will need to discuss with DO and Telemetry teams.